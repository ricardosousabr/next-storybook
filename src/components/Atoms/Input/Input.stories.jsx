import Input from "./index.jsx";

export default {
  title: "Components/Input",
  Component: Input,
  parameters: {
    backgrounds: {
      values: [
        { name: "dark", value: "#000000" },
        { name: "white", value: "#ffffff" },
        { name: "purple", value: "#b220d7" },
      ],
    },
  },
};

export const Primary = (args) => <Input {...args} />;
Primary.args = {
  padding: "",
  borderRadius: "",
  value: "",
  placeholder: "",
  type: "text",
};
