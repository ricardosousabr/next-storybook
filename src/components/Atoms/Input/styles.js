import { styled } from "styled-components";

export const InputStyle = styled.input`
  border-radius: ${(props) => props.borderRadius};
  padding: ${(props) => props.padding} ;
`