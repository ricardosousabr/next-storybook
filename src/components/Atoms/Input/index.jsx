import PropTypes from "prop-types";
import {InputStyle} from "./styles.js"

export default function Input({ type, placeholder, value, borderRadius, padding }) {
  return <InputStyle padding={padding} borderRadius={borderRadius} value={value} placeholder={placeholder} type={type} />
}

Input.prototype = {
  padding: PropTypes.string,
  borderRadius: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf(["button", "checkbox", "color", "date", "datetime-local", "email", "file", "hidden", "image", "month", "number", "password", "radio", "range", "reset", "search", "submit", "tel", "text", "time", "url", "week"])
}