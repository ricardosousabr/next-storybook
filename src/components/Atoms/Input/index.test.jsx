import renderer from "react-test-renderer";
import Input from "./index";

test("render input", () => {
  const tree = renderer.create(<Input type="text"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
