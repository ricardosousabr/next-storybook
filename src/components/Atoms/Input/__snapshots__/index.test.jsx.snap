// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input 1`] = `
<input
  className="styles__InputStyle-sc-xbforv-0 gRnhN"
  type="text"
/>
`;
